Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -92,71 +242,268 @@ void SkylanderPortalWindow::CreateMainWindow()
+    hbox_skylander->addWidget(button);
     hbox_skylander->addWidget(label_skyname);
     hbox_skylander->addWidget(m_edit_skylanders[i]);
-    hbox_skylander->addWidget(clear_btn);
-    hbox_skylander->addWidget(create_btn);
-    hbox_skylander->addWidget(load_btn);
 
     vbox_group->addLayout(hbox_skylander);
   }
+  m_slot_radios[0]->setChecked(true);
 
   m_group_skylanders->setLayout(vbox_group);
   scroll_area->setWidget(m_group_skylanders);
   scroll_area->setWidgetResizable(true);
   m_group_skylanders->setVisible(Config::Get(Config::MAIN_EMULATE_SKYLANDER_PORTAL));
-  main_layout->addWidget(scroll_area);
-  setLayout(main_layout);
+  slot_layout->addWidget(scroll_area);
 
-  UpdateEdits();
+  slot_group->setLayout(slot_layout);
+  slot_group->setMaximumWidth(350);
+
+  return slot_group;
 }
 
-void SkylanderPortalWindow::OnEmulationStateChanged(Core::State state)
+QGroupBox* SkylanderPortalWindow::CreateSearchGroup()
 {
-  const bool running = state != Core::State::Uninitialized;
+  m_skylander_list = new QListWidget;
+
+  m_skylander_list->setMinimumWidth(200);
+  connect(m_skylander_list, &QListWidget::itemDoubleClicked, this,
+          &SkylanderPortalWindow::LoadSelected);
+
+  auto* main_group = new QGroupBox();
+  auto* main_layout = new QVBoxLayout();
+
+  auto* header_group = new QGroupBox();
+  auto* header_layout = new QHBoxLayout();
+
+  header_layout->addWidget(new QLabel(tr("Skylander Collection Path:")));
+  m_path_edit = new QLineEdit;
+  header_layout->addWidget(m_path_edit);
+  m_path_select = new QPushButton(tr("Choose"));
+  connect(m_path_edit, &QLineEdit::editingFinished, this,
+          &SkylanderPortalWindow::OnCollectionPathChanged);
+  connect(m_path_select, &QAbstractButton::clicked, this,
+          &SkylanderPortalWindow::SelectCollectionPath);
+  header_layout->addWidget(m_path_select);
+
+  header_group->setLayout(header_layout);
+  main_layout->addWidget(header_group);
+
+  auto* search_bar_layout = new QHBoxLayout;
+  m_sky_search = new QLineEdit;
+  m_sky_search->setClearButtonEnabled(true);
+  connect(m_sky_search, &QLineEdit::textChanged, this, &SkylanderPortalWindow::RefreshList);
+  search_bar_layout->addWidget(new QLabel(tr("Search:")));
+  search_bar_layout->addWidget(m_sky_search);
+  main_layout->addLayout(search_bar_layout);
+
+  auto* search_group = new QGroupBox();
+  auto* search_layout = new QHBoxLayout();
+
+  auto* search_filters_group = new QGroupBox();
+  auto* search_filters_layout = new QVBoxLayout();
+
+  auto* search_checkbox_group = new QGroupBox(tr("Game"));
+  auto* search_checkbox_layout = new QVBoxLayout();
+
+  for (int i = 0; i < 5; i++)
+  {
+    QCheckBox* checkbox = new QCheckBox(this);
+    checkbox->setChecked(true);
+    connect(checkbox, &QCheckBox::toggled, this, &SkylanderPortalWindow::RefreshList);
+    m_game_filters[i] = checkbox;
+    search_checkbox_layout->addWidget(checkbox);
+  }
+  m_game_filters[0]->setText(tr("Spyro's Adventure"));
```

Comment:
If you move `G_OTHER` at the end of `enum Game`, you could replace `m_game_filters[0..5]` with `m_game_filters[SPYROS_ADV..SUPERCHARGERS]`.
Something similar is possible for `m_element_filter`.